mockatests/drivers/test-scheduler.c

```diff
@@ -0,0 +1,1315 @@
+{
+  // Setting a new higher value will be ignored
+  scheduler_t s;
+  s.max_timeout.tv_sec = 430;
+  struct timeval timeout = {.tv_sec = 458 };
+  scheduler_set_max_timeout(&s, timeout);
+  assert_int_equal(s.max_timeout.tv_sec, 430);
+}
+
+void
+test_scheduler_register_event_null_callback(void **state)
+{
+  // Callback cannot be NULL
+  scheduler_t s;
+  scheduler_initialize(&s);
+  char mode = SCHEDULER_POLL_TIMEOUT;
+  int fd = 0;
+  struct timeval timeout = {};
+  event_cb_t cb = NULL;
+  void *private = NULL;
+
+  const int r = scheduler_register_event(&s, mode, fd, timeout, cb, private);
+  assert_int_equal(r, -EINVAL);
+}
+
+void
+test_scheduler_register_event_bad_mode(void **state)
+{
+  // Bad mode
+  scheduler_t s;
+  scheduler_initialize(&s);
+  char mode = 0;
+  int fd = 0;
+  struct timeval timeout = {};
+  event_cb_t cb = &fake_event_cb;
+  void *private = NULL;
+
+  const int r = scheduler_register_event(&s, mode, fd, timeout, cb, private);
+  assert_int_equal(r, -EINVAL);
+}
+
+void
+test_scheduler_register_multiple_events(void **state)
+{
+  scheduler_t s;
+  scheduler_initialize(&s);
+
+  char mode = SCHEDULER_POLL_TIMEOUT;
+  int fd = 0;
+  struct timeval timeout = {};
+  event_cb_t cb = &fake_event_cb;
+  void *private = NULL;
+
+  /* Event queue starts off empty */
+  assert_int_equal(event_queue_length(&s), 0);
+
+  /* Event queue starts now has one event */
+  const int event_id1 = scheduler_register_event(&s, mode, fd, timeout, cb, private);
+  assert_int_equal(event_queue_length(&s), 1);
+
+  /* Event queue starts now has two events */
+  const int event_id2 = scheduler_register_event(&s, mode, fd, timeout, cb, private);
+  assert_int_equal(event_queue_length(&s), 2);
+
+  /* The two event IDs we were given are different */
+  assert_int_not_equal(event_id1, event_id2);
+}
+
+void
+test_scheduler_register_event_populates_event(void **state)
+{
+  /* scheduler_register_event will fill out all fields in event */
+  scheduler_t s;
+  scheduler_initialize(&s);
+
+  char mode = SCHEDULER_POLL_TIMEOUT;
+  int fd = 458;
+  struct timeval timeout = { .tv_sec = 575, .tv_usec = 599 };
+  event_cb_t cb = &fake_event_cb;
+  int private = 430;
+
+  fake_gettimeofday = (struct timeval){ .tv_sec = 2, .tv_usec = 3};
+  const int event_id1 = scheduler_register_event(&s, mode, fd, timeout, cb, &private);
+
+  const event_t* e = list_first_entry(&s.events, event_t, next);
+
+  assert_int_equal(e->id, event_id1);
+  assert_int_equal(e->mode, mode);
+  assert_int_equal(e->fd, fd);
+  assert_int_equal(e->timeout.tv_sec, timeout.tv_sec);
+  assert_int_equal(e->timeout.tv_usec, timeout.tv_usec);
+  assert_ptr_equal(e->cb, &fake_event_cb);
+  assert_ptr_equal(e->private, &private);
+  assert_ptr_equal(e->masked, 0);
+
+  assert_int_equal(e->deadline.tv_sec,  fake_gettimeofday.tv_sec + timeout.tv_sec);
+  assert_int_equal(e->deadline.tv_usec, fake_gettimeofday.tv_usec + timeout.tv_usec);
+}
+
+void
+test_scheduler_set_timeout_inf(void **state)
```

Comment:
`test_scheduler_register_event_timeout_inf` ?

Reply:
Added in b5a2f60.